### PR TITLE
refactor: add type hints to saferound.py

### PR DESCRIPTION
**Pull request by tcrasset**
- chore(processor): run formatter
- refactor: add type hints and refactor code in saferound.py
- fix: call saferound with list instead of np.ndarray

### CODE REVIEW

File: avatars/lib/saferound.py [hunk 1]
```diff
@@ -47,34 +79,47 @@ def saferound(iterable, places, strategy=DIFFERENCE, rounder=round, topline=None
             - strategy is not valid
             - values are not all floats
     """
-    assert isinstance(places, int)
-    assert strategy in [SMALLEST, LARGEST, DIFFERENCE]
```
Comment (tas17) [inline avatars/lib/saferound.py:51]: Are you sure an `Enum` is checked at runtime ?

File: avatars/lib/saferound_test.py [hunk 2]
```diff
@@ -1,116 +1,114 @@
...
-    with pytest.raises(
-        AssertionError
-    ):
-        saferound(setup.in_list, 2.5)
-
-def test_error_bad_strategy(setup):
-    with pytest.raises(
-        AssertionError
-    ):
-        saferound(setup.in_list, 2, strategy='foo')
```
Comment (tas17) [inline avatars/lib/saferound_test.py:75]: This test was testing validation is done at runtime (see my other comment). Why did you remove it ?

**Review by olivierabz — COMMENTED** [review]
Looks good! Thanks for this cleaning

**Review by olivierabz — APPROVED** [review]

File: avatars/lib/saferound.py (hunk 1 above)
Comment (tcrasset) [reply avatars/lib/saferound.py:51]: No, none of them are, but this is a lib that is used internally in our code, which we typecheck.
It though about keeping them in, but that would have resulted in many `if-else` checks, as `bandit` complains about asserts
